flow/include/flow/EncryptUtils.h

```diff
@@ -37,14 +37,19 @@
 
 #define SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID -2
```

Comment:
Shall we have a dedicated domain name for the system domain?

Reply:
Prior to this change we have two type of encryption domains:
1. FDB metadata encryption domain - it encrypts FDB internal database metadata.
2. FDB tenant encryption domain

My belief is even after this change we retain two categories of encryption domains
1. FDB metadata encryption domain - it encrypts FDB internal database metadata.
2. FDB tenant encryption domain 
3. FDB default encyrption domain - data which isn't FDB internal database, but Tenant information isn't available. 

To me reusing #1 as `default domain` dilutes the encryption domain boundary by containing either `FDB internal database metadata` OR `unknown tenant data`; from security perspective it seems reasonable to ensure this partition. Further, as/when we enable `Encryption consistency checker` we can `assert` that above defined domain boundaries are respected across all encryption participant FDB processes.

Reply:
I have no strong opinion one way or the other. 
However, for completeness it feels OK to have `domain_name` 1:1 mapping for both `system` and `default` encryption domains. 
Thanks